experimental/rocm/array_command_buffer.c

```diff
@@ -0,0 +1,313 @@
+                                                                       iree_hal_executable_layout_t *executable_layout,
+                                                                       iree_host_size_t offset, const void *values,
+                                                                       iree_host_size_t values_length) {
+  return iree_make_status(IREE_STATUS_UNIMPLEMENTED, "need rocm implementation");
+}
+
+static iree_status_t iree_hal_rocm_array_command_buffer_push_descriptor_set(
+  iree_hal_command_buffer_t *base_command_buffer, iree_hal_executable_layout_t *executable_layout, uint32_t set,
+  iree_host_size_t binding_count, const iree_hal_descriptor_set_binding_t *bindings) {
+  iree_hal_rocm_array_command_buffer_t *command_buffer = iree_hal_rocm_array_command_buffer_cast(base_command_buffer);
+  for (iree_host_size_t i = 0; i < binding_count; i++) {
+    uint32_t arg_index = bindings[i].binding;
+    assert(arg_index < max_binding_count && "binding index larger than the max expected.");
+    hipDeviceptr_t device_ptr =
+      iree_hal_rocm_buffer_device_pointer(iree_hal_buffer_allocated_buffer(bindings[i].buffer)) +
+      iree_hal_buffer_byte_offset(bindings[i].buffer) + bindings[i].offset;
+    *((hipDeviceptr_t *)command_buffer->current_descriptor[arg_index]) = device_ptr;
+  }
+  return iree_ok_status();
+}
+
+static iree_status_t iree_hal_rocm_array_command_buffer_bind_descriptor_set(
+  iree_hal_command_buffer_t *base_command_buffer, iree_hal_executable_layout_t *executable_layout, uint32_t set,
+  iree_hal_descriptor_set_t *descriptor_set, iree_host_size_t dynamic_offset_count,
+  const iree_device_size_t *dynamic_offsets) {
+  return iree_make_status(IREE_STATUS_UNIMPLEMENTED, "need rocm implementation");
+}
+
+static iree_status_t iree_hal_rocm_array_command_buffer_dispatch(iree_hal_command_buffer_t *base_command_buffer,
+                                                                 iree_hal_executable_t *executable, int32_t entry_point,
+                                                                 uint32_t workgroup_x, uint32_t workgroup_y,
+                                                                 uint32_t workgroup_z) {
+  iree_hal_rocm_array_command_buffer_t *command_buffer = iree_hal_rocm_array_command_buffer_cast(base_command_buffer);
+  iree_hal_rocm_array_command_buffer_cast(base_command_buffer);
+
+  int32_t block_size_x, block_size_y, block_size_z;
+  IREE_RETURN_IF_ERROR(
+    iree_hal_rocm_native_executable_block_size(executable, entry_point, &block_size_x, &block_size_y, &block_size_z));
+  int size = command_buffer->kernelArray.total_size;
+  void *config[] = {HIP_LAUNCH_PARAM_BUFFER_POINTER, &command_buffer->current_descriptor[0],
+                    HIP_LAUNCH_PARAM_BUFFER_SIZE, &size, HIP_LAUNCH_PARAM_END};
+  // Add Kernel Information into KernelArray.
+  command_buffer->kernelArray.kernels[command_buffer->kernelArray.num_of_kernels].func =
```

Comment:
You don't want an array of kernel anymore right? Looks like you don't use it, you should be able to just clean that up.